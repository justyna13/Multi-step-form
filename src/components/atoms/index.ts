export { FormStep } from "@/components/atoms/FormStep/FormStep.tsx";
export { Button } from "@/components/atoms/Button/Button.tsx";
export { Input } from "@/components/atoms/Input/Input.tsx";
export { Header } from "@/components/atoms/Header/Header.tsx";
