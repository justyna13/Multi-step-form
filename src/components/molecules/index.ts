export { FormNavigation } from "@/components/molecules/FormNavigation/FormNavigation.tsx";
export { Footer } from "@/components/molecules/Footer/Footer.tsx";
export { FormInput } from "@/components/molecules/FormInput/FormInput.tsx";
export { FormSummary } from "@/components/molecules/FormSummary/FormSummary.tsx";
