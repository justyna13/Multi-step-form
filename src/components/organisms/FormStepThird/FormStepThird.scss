.form-step--third {
  .step-bottom {
    display: flex;
    justify-content: space-between;
    button {
      margin-top: 48px;
      margin-bottom: 12px;
    }
  }
}
