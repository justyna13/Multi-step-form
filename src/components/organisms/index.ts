export { FormStepFirst } from '@/components/organisms/FormStepFirst/FormStepFirst.tsx';
export { FormStepSecond } from '@/components/organisms/FormStepSecond/FormStepSecond.tsx';
export { FormStepThird } from '@/components/organisms/FormStepThird/FormStepThird.tsx';
export { FormStepFourth } from '@/components/organisms/FormStepFourth/FormStepFourth.tsx';
export { FormConfirmation } from '@/components/organisms/FormConfirmation/FormConfirmation.tsx';
