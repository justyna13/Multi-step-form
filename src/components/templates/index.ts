export { HomePageTemplate } from '@/components/templates/HomePageTemplate/HomePageTemplate.tsx';
