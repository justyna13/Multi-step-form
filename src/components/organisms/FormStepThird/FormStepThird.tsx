import { useFormContext } from 'react-hook-form';

import { Header } from '@/components/atoms';
import { Button } from '@/components/atoms';
import { AddOn } from '@/components/molecules';
import '@/components/organisms/FormStepThird/FormStepThird.scss';
import { availableAddOns } from '@/constants/data.tsx';
import { MultiStepFormActionType, useMultiStepFormContext } from '@/store';

interface IFormStepThird {
  testid: string;
  className?: string;
  onBtnNextClicked: () => void;
  onBtnPrevClicked: () => void;
}

export type SingleAddOn = {
  id: number;
  name: string;
  pricePerMonth: number;
  description: string;
  fieldName: string;
};

export const FormStepThird: React.FC<IFormStepThird> = ({
  testid,
  onBtnPrevClicked,
  onBtnNextClicked
}: IFormStepThird) => {
  const { getValues } = useFormContext();
  const { dispatch } = useMultiStepFormContext();

  const saveData = () => {
    const stepFields = getValues();
    dispatch({
      type: MultiStepFormActionType.UPDATE_STEP_ADDITIONAL,
      payload: stepFields
    });
    onBtnNextClicked();
  };

  return (
    <div className="form-step--third" data-testid={testid}>
      <Header
        testid={'multiStepForm-step-third-ttl'}
        title={'Pick add-ons'}
        description={'Add-ons help enhance your gaming experience. '}
      />

      {availableAddOns.map((item) => (
        <AddOn
          name={`addOns.${item.fieldName}`}
          testid={`add-on-${item.id}`}
          item={item}
          key={item.id}
        />
      ))}

      <div className="step-bottom">
        <Button
          type="button"
          variant="prev"
          buttonText="Go Back"
          onClick={onBtnPrevClicked}
        />
        <Button
          type="button"
          variant="next"
          buttonText="Next Step"
          onClick={saveData}
        />
      </div>
    </div>
  );
};
