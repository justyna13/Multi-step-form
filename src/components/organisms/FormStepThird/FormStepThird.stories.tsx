import { Meta, StoryObj } from '@storybook/react';
import { FormProvider, useForm } from 'react-hook-form';

import { FormStepThird } from '@/components/organisms';
import { MultiStepFormProvider } from '@/store';

const meta = {
  title: 'Organisms/FormStepThird',
  component: FormStepThird,
  tags: ['autodocs'],
  decorators: [
    (Story, context) => {
      const methods = useForm();

      return (
        <MultiStepFormProvider>
          <FormProvider {...methods}>
            <Story>
              <FormStepThird {...context.args} />
            </Story>
          </FormProvider>
        </MultiStepFormProvider>
      );
    }
  ]
} satisfies Meta<typeof FormStepThird>;

export default meta;
type Story = StoryObj<typeof meta>;

export const DefaultFormStepThird: Story = {
  args: {
    testid: 'testid',
    onBtnNextClicked: () => {},
    onBtnPrevClicked: () => {}
  }
};
