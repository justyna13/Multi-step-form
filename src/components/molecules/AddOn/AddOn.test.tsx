import { render, screen } from '@testing-library/react';
import { describe, expect } from 'vitest';

import { AddOn } from '@/components/molecules';
import { SingleAddOn } from '@/components/organisms/FormStepThird/FormStepThird.tsx';

const exampleAddOn: SingleAddOn = {
  id: 1,
  name: 'test',
  pricePerMonth: 12,
  description: 'test desc',
  fieldName: 'onlineService'
};

const mockClickFn = vi.fn();

describe('AddOn', () => {
  it('should match snapshot', () => {
    const view = render(
      <AddOn testid={'test'} name="test" item={exampleAddOn} />
    );

    expect(view).toMatchSnapshot();
  });

  it('should render disabled', () => {
    const view = render(
      <AddOn testid={'test'} name="test" item={exampleAddOn} disabled={true} />
    );
    const input = view.container.querySelector('#add-on-1');
    expect(input).toHaveAttribute('disabled', '');
  });

  it('should render item texts', () => {
    render(<AddOn testid={'test-addon'} name="test" item={exampleAddOn} />);

    const addOnWrapper = screen.getByTestId('test-addon');

    expect(addOnWrapper).toHaveTextContent(exampleAddOn.name);
    expect(addOnWrapper).toHaveTextContent(exampleAddOn.description);
    expect(addOnWrapper).toHaveTextContent(`${exampleAddOn.pricePerMonth}`);
  });

  it('should react onClick', () => {
    render(<AddOn testid={'test'} name="test" item={exampleAddOn} />);

    // const addOnWrapper = screen.getByTestId('test');
    expect(mockClickFn).toBeCalledTimes(0);
    // todo
  });
});
