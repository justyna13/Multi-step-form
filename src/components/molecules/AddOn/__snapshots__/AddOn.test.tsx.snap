// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddOn > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="add-on  "
        data-testid="test"
      >
        <div
          class="checkbox-wr add-on__inner"
        >
          <input
            id="add-on-1"
            name="test"
            type="checkbox"
          />
          <label
            for="add-on-'1"
          >
            <div
              class="add-on__details"
            >
              <div>
                <p
                  class="add-on__ttl"
                >
                  test
                </p>
                <p
                  class="add-on__desc"
                >
                  test desc
                </p>
              </div>
              <div
                class="add-on__price"
              >
                +$
                12
                /mo
              </div>
            </div>
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="add-on  "
      data-testid="test"
    >
      <div
        class="checkbox-wr add-on__inner"
      >
        <input
          id="add-on-1"
          name="test"
          type="checkbox"
        />
        <label
          for="add-on-'1"
        >
          <div
            class="add-on__details"
          >
            <div>
              <p
                class="add-on__ttl"
              >
                test
              </p>
              <p
                class="add-on__desc"
              >
                test desc
              </p>
            </div>
            <div
              class="add-on__price"
            >
              +$
              12
              /mo
            </div>
          </div>
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
